nixos/modules/services/web-servers/apache-httpd/default.nix

```diff
@@ -126,10 +126,17 @@ let
     </IfModule>
   '';
 
-  luaSetPaths = ''
+  luaSetPaths = let
+    luaversion =
+      if builtins.hasAttr "lua" cfg.package.lua5 then
+        cfg.package.lua5.lua.luaversion
+      else
+        cfg.package.lua5.luaversion;
+    in
+  ''
     <IfModule mod_lua.c>
-      LuaPackageCPath ${cfg.package.lua5}/lib/lua/${cfg.package.lua5.lua.luaversion}/?.so
-      LuaPackagePath  ${cfg.package.lua5}/share/lua/${cfg.package.lua5.lua.luaversion}/?.lua
+      LuaPackageCPath ${cfg.package.lua5}/lib/lua/${luaversion}/?.so
+      LuaPackagePath  ${cfg.package.lua5}/share/lua/${luaversion}/?.lua
     </IfModule>
```

Comment:
I think we had some refactoring. There is no lua version that has a lua attribute.
Just make it unconditional: `cfg.package.lua5.luaversion`

Reply:
I just tested it out and I dont see anything changed.
What I mean, why this patch is needed is the following: 

Lets say you have this nixos config:
```nix
{
  services.httpd = {
      enable = true;
      package = pkgs.apacheHttpd.override {
        luaSupport = true;
        lua5 = pkgs.lua5_3.withPackages (ps: with ps; [ luafilesystem ] );
      };
    };
}
```

Here I say that I want to have my apache to use a lua packaged with the
`luafilesystem` module so that I can `require` that in scripts to render my http
responses. There, the set that gets assigned to `lua5 ` does not have a
`luaversion` attribute, rather it has a `lua` attribute wherein lies a
`luaversion` attribute. If I dont package additional modules, then I dont have
that `lua` attribute in between and rather directly have to use `luaversion`
directly.

So as far as I tell the code is still needed. Please correct me if I am wrong.

I guess I can make it a bit shorter by saying something like:

```nix
luaversion = (cfg.package.lua5.lua.luaversion or cfg.package.lua5.luaversion);
```

Reply:
Ok. Now I see the usecase. Can you add this to a comment why this is required?

Reply:
It was also explained in the commit message, but I have extended the message to include this example.